source/uwp/Renderer/lib/AdaptiveColumnRenderer.cpp

```diff
@@ -3,168 +3,135 @@
+            {
+                currentRtl = containerRtl;
+                renderContext.Rtl(currentRtl);
+                updatedRtl = true;
+            }
 
-        ComPtr<IReference<bool>> containerRtl;
-        RETURN_IF_FAILED(adaptiveColumn->get_Rtl(&containerRtl));
+            if (currentRtl != nullptr)
+            {
+                columnPanel.FlowDirection(currentRtl.Value() ? rtxaml::FlowDirection::RightToLeft : rtxaml::FlowDirection::LeftToRight);
+            }
 
-        bool updatedRtl = false;
-        if (containerRtl != nullptr)
-        {
-            currentRtl = containerRtl;
-            RETURN_IF_FAILED(renderContext->put_Rtl(currentRtl.Get()));
-            updatedRtl = true;
-        }
+            rtom::ContainerStyle containerStyle{};
 
-        if (currentRtl)
-        {
-            boolean rtlValue;
-            RETURN_IF_FAILED(currentRtl->get_Value(&rtlValue));
+            // TODO: do I even need this if (const auto..) ?
+            if (const auto columnAsContainerBase = adaptiveColumn.try_as<rtom::IAdaptiveContainerBase>())
+            {
+                containerStyle = ::AdaptiveCards::Rendering::Uwp::XamlHelpers::HandleStylingAndPadding(columnAsContainerBase,
+                                                                                                          columnBorder,
+                                                                                                          renderContext,
+                                                                                                          renderArgs);
+            }
 
-            RETURN_IF_FAILED(columnPanelAsFrameworkElement->put_FlowDirection(rtlValue ? FlowDirection_RightToLeft :
-                                                                                         FlowDirection_LeftToRight));
-        }
+            auto parentElement = renderArgs.ParentElement();
 
-        ComPtr<IAdaptiveContainerBase> columnAsContainerBase;
-        RETURN_IF_FAILED(adaptiveColumn.As(&columnAsContainerBase));
+            auto newRenderArgs =
+                winrt::make<rtrender::implementation::AdaptiveRenderArgs>(containerStyle, parentElement, renderArgs);
 
-        ABI::AdaptiveCards::ObjectModel::Uwp::ContainerStyle containerStyle;
-        RETURN_IF_FAILED(
-            XamlHelpers::HandleStylingAndPadding(columnAsContainerBase.Get(), columnBorder.Get(), renderContext, renderArgs, &containerStyle));
+            auto childItems = adaptiveColumn.Items();
 
-        ComPtr<IFrameworkElement> parentElement;
-        RETURN_IF_FAILED(renderArgs->get_ParentElement(&parentElement));
-        ComPtr<IAdaptiveRenderArgs> newRenderArgs;
-        RETURN_IF_FAILED(MakeAndInitialize<AdaptiveRenderArgs>(&newRenderArgs, containerStyle, parentElement.Get(), renderArgs));
+            ::AdaptiveCards::Rendering::Uwp::XamlBuilder::BuildPanelChildren(
+                childItems, columnPanel, renderContext, newRenderArgs, [](rtxaml::IUIElement) {});
```

Comment:
Using `auto&&` makes it a more explicit discard and avoids a copy.

```suggestion
                childItems, columnPanel, renderContext, newRenderArgs, [](auto&&) {});
```

Reply:
@oldnewthing can I please ask, what's the difference between **auto&** and **auto&&**?